src/log-analytics/azext_loganalytics/_help.py

```diff
@@ -7,22 +7,29 @@
 
 # pylint: disable=line-too-long
 
+helps['monitor log-analytics'] = """
+    type: group
+    short-summary: Commands for querying data in Log Analytics workspaces.
+"""
+
 helps['monitor log-analytics query'] = """
     type: command
     short-summary: Query a Log Analytics workspace.
     parameters:
-        - workspace: --workspace -w
-          type: string
-          short-summary: GUID of the Log Analytics workspace.
-        - analytics-query: --analytics-query
-          type: string
-          short-summary: Query to execute over the Log Analytics data.
-        - timespan: --timespan -t
-          type: string
-          short-summary: Timespan over which to query data. Defaults to all data.
-        - workspaces: --workspaces
-          type: array
-          short-summary: Additional workspaces to union data for querying. Specify additional workspace IDs separated by commas.
+      - name: --workspace
+        type: string
+        short-summary: GUID of the Log Analytics workspace.
+      - name: --analytics-query
+        type: string
+        short-summary: Query to execute over the Log Analytics data.
+      - name: --timespan
```

Comment:
I believe the CI test you are failing is because you don't have the correct parameter help names for some of your parameters. In[ _params.py](https://github.com/Azure/azure-cli-extensions/blob/90a71cac2d7b6915fa5d73d8f2d7e11699e6c9d7/src/log-analytics/azext_loganalytics/_params.py), you have:

- your `workspace` parameter has the options `options_list=['--workspace', '-w']`. However, in your help.py your parameter name is `--workspace` instead of `--workspace -w`
- your `timespan` parameter has the options `options_list=['--timespan', '-t']`. However, in your help.py your parameter name is `--timespan` instead of `--timespan -t`

So to solve this problem, you can: 
- use the old help.py yaml parameter names before your change
- or remove the help entry for your **all** your parameters, so the help can be generated from _params.py.

Reply:
The old parameter names don't seem to parse any more, seems like some validation changed on the key fields?

Additionally, I [tried using the naming](https://github.com/Azure/azure-cli-extensions/pull/316/commits/65753c4ea35fd5c90585f5555a2b2fbc6d3bde04) you recommend after finding [this issue[(https://github.com/Microsoft/knack/issues/85). It didn't work for me, the validation still failed CI. 

Conveniently, if I remove the parameters which fail name validation from the help file, they autopopulate the parameter help correctly. The remaining parameters pass name validation but the help generation doesn't always work, so I added a string in the help file to populate the --analytics-query parameter properly.

Reply:
Yes, if help for a parameter is missing in the Yaml file, it would be generated from the help argument in _params.py, as mentioned above.

Hmm that's strange that using the old naming didn't work. 

I am happy though that you were able to resolve this issue.